src/plugins/intel_gpu/src/graph/include/program_node.h

```diff
@@ -415,10 +418,18 @@ struct program_node {
         cur_id = 0;
     }
 
-    format::type get_required_input0() const { return required_input0; }
-    format::type get_required_output() const { return required_output; }
-    void set_required_input0(format::type type) { required_input0 = type; }
-    void set_required_output(format::type type) { required_output = type; }
+    std::vector<format::type> get_preferred_input_fmts() const { return preferred_input_fmts; }
+    std::vector<format::type> get_preferred_output_fmts() const { return preferred_output_fmts; }
+    format::type get_preferred_input_fmt(size_t idx = 0) const {
+        return (idx < preferred_input_fmts.size()) ? preferred_input_fmts.at(idx) : format::any;
+    }
+    format::type get_preferred_output_fmt(size_t idx = 0) const {
+        return (idx < preferred_output_fmts.size()) ? preferred_output_fmts.at(idx) : format::any;
```

Comment:
do we need this condition `idx < preferred_output_fmts.size()`? Isn't it possible to replace it with exception?

Reply:
Reverted. get_preferred_fmts functions are called even if ENABLE_ONEDNN_FOR_GPU is disabled.
So it needs to return any which means no preferred format.